crates/common/src/signer/mod.rs

```diff
@@ -0,0 +1,122 @@
+use std::{
+    error::Error,
+    fmt::{self, LowerHex},
+};
+
+use eyre::Context;
+use serde::{Deserialize, Serialize};
+use ssz_derive::{Decode, Encode};
+use tree_hash::TreeHash;
+
+pub mod schemes;
+#[allow(clippy::module_inception)]
+mod signer;
+
+pub use schemes::{bls::BlsSecretKey, ecdsa::EcdsaSecretKey};
+pub use signer::{ConsensusSigner, Signer};
+
+pub type Pubkey<T> = <T as SecretKey>::PublicKey;
```

Comment:
We'd like to avoid this type alias and stick to concrete types (opinionated is not that bad in our case).

Reply:
Resolved in [eba97b0](https://github.com/Commit-Boost/commit-boost-client/pull/87/commits/eba97b0ca03b140bbfdc9f4b7b8ad4adf382776b)